staging/src/kubevirt.io/api/usb/v1alpha1/types.go

```diff
@@ -0,0 +1,50 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NodeConfig represents a subset of Node resources that we want to expose
+// to virtual machines.
+//
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +genclient
+// +genclient:nonNamespaced
+type NodeConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              NodeConfigSpec   `json:"spec" valid:"required"`
+	Status            NodeConfigStatus `json:"status,omitempty"`
+}
+
+type NodeConfigSpec struct {
+	USB USB `json:"usb,omitempty"`
+}
+
+// USB defines a group of USB devices based on its selectors and combine them under a single
+// resource name, to be allocated to the VM
+type USB struct {
+	// Identifies the list of USB host devices.
+	// e.g: kubevirt.io/storage, kubevirt.io/bootable-usb, etc
+	ResourceName string `json:"resourceName"`
+	// +listType=atomic
+	USBHostDevices []USBHostDevices `json:"usbHostDevices,omitempty"`
+}
+
+type USBHostDevices struct {
+	// The vendor:product of the devices we want to select.
+	// e.g: "0951:1666"
+	SelectByVendorProduct string `json:"selectByVendorProduct"`
```

Comment:
Using `struct` here instead of string might help in understanding the API better and future validation.

Reply:
I don't mind changing it. I'm used to this format due `virtctl usbredir` wrapping `usbredirect` binary for client-side USB redirection. Also, this is exactly how [pciVendorSelector](https://kubevirt.io/user-guide/virtual_machines/host-devices/#host-preparation-for-pci-passthrough) works.